README.md

```diff
@@ -4,130 +4,183 @@
 
 # Swift for TensorFlow
 
-Welcome to the Swift for TensorFlow development community! For discussions, join the [swift@tensorflow.org mailing list](https://groups.google.com/a/tensorflow.org/d/forum/swift).
+> Swift for TensorFlow: No barriers.
+
+Swift for TensorFlow (aka S4TF) is a next-generation system for machine
+learning, incorporating the latest research across: machine learning,
+compilers, differentiable programming, systems design, and beyond. This project
+is approaching version _0.2_; it is neither feature complete nor
+production-ready. But it is ready for _pioneers_ to try it for your own
+projects, give us feedback, and help shape the future!
+
+The S4TF project is currently focusing on 2 kinds of users:
+
+ 1. **Advanced ML researchers** who are limited by current ML frameworks.
+    S4TF's advantages include a seamless integration with a modern general-purpose
+    language, allowing for more dynamic and sophisticated models. Fast
+    abstractions can be developed "in user-space" (as opposed to in C/C++
+    aka "framework-space"), resulting in modular APIs that can be easily
+    customized.
+
+ 2. **ML learners** who are just getting started with machine learning. Thanks
+    to Swift's support for quality tooling (e.g. context-aware autocomplete),
+    Swift for TensorFlow can be one of the most productive ways to get started
+    learning the fundamentals of machine learning.
+
+## Getting started
+
+ - **Google Colaboratory**: The fastest way to get started is to try out S4TF right in your
+   browser. Just open up our [getting started
+   notebook](https://colab.research.google.com/github/tensorflow/swift-tutorials/blob/master/iris/swift_tensorflow_tutorial.ipynb) (or start from a
+   [blank notebook](https://colab.research.google.com/github/tensorflow/swift/blob/master/notebooks/blank_swift.ipynb))!
+   Read more in our [usage guide](Usage.md).
+
+ - **Install locally**: you can [download a pre-built Swift for TensorFlow
+   package](Installation.md). After installation, you can follow these
+   [step-by-step instructions](Usage.md) to build and execute a Swift script
+   on your computer.
+
+ - **Compile from source**: If you'd like to customize S4TF or even contribute
+   back, follow our [instructions on building the S4TF compiler from
+   source](https://github.com/apple/swift/tree/tensorflow).
+
+Please do join the
+[swift@tensorflow.org mailing list](https://groups.google.com/a/tensorflow.org/d/forum/swift)
+to hear the latest announcements, get help, and share your thoughts!
+
+## Why Swift for TensorFlow?
 
 Swift for TensorFlow is a new way to develop machine learning models. It
 gives you the power of
 [TensorFlow](https://www.tensorflow.org) directly integrated into the
```

Comment:
I think we should mention "protocol-oriented differentiable programming" since the vision here is bigger than ML and TensorFlow.
Maybe here: "the power of `TensorFlow` and protocol-oriented differentiable programming"
But I'm not sure.

Reply:
Hmm, I think this is becoming quite wordy, and trying to wedge that in seems a little unnecessary given that the next paragraph and code demonstration cover the concept pretty effectively. Does that make sense?

Reply:
SGTM!